### PR TITLE
use LinkedList instead of List in PriorityProducerConsumer

### PR DESCRIPTION
**Pull request by heejaechang**
simple clean up. PriorityProducerConsumer was using List but removed element from head. but never took advantage of indexer so changed it to use LinkedList.

### CODE REVIEW

File: src/Core/Impl/Extensions/TaskCompletionSourceExtensions.cs [hunk 1]
```diff
@@ -18,8 +18,10 @@
 
 namespace Microsoft.Python.Core {
     public static class TaskCompletionSourceExtensions {
+        private const int NO_DELAY = -1;
```
Comment (AlexanderSher) [inline src/Core/Impl/Extensions/TaskCompletionSourceExtensions.cs]: AFAIK, we don't use all caps notation for constants.

Comment (heejaechang) [reply src/Core/Impl/Extensions/TaskCompletionSourceExtensions.cs]: @AlexanderSher how should I change it? no_delay then?

Comment (heejaechang) [reply src/Core/Impl/Extensions/TaskCompletionSourceExtensions.cs]: changed to lower case.

File: src/Core/Impl/Extensions/TaskCompletionSourceExtensions.cs [hunk 2]
```diff
@@ -18,8 +18,10 @@
 
 namespace Microsoft.Python.Core {
     public static class TaskCompletionSourceExtensions {
+        private const int no_delay = -1;
```
Comment (jakebailey) [inline src/Core/Impl/Extensions/TaskCompletionSourceExtensions.cs]: Too late, but probably should have been `NoDelay` for style consistency.

**Pull request merged**